gcm4/src/test/java/plugins/personproperties/datamanagers/AT_PersonPropertyDataManager.java

```diff
@@ -1334,6 +1333,158 @@ public boolean equals(Object obj) {
 
 	}
 
+	@Test
+	@UnitTestMethod(target = PersonPropertiesDataManager.class, name = "getEventFilterForPersonPropertyUpdateEvent", args = {
+			PersonPropertyId.class, Object.class, boolean.class })
+	public void testGetEventFilterForPersonPropertyUpdateEvent_propertyId_object() {
+
+		RandomGenerator randomGenerator = RandomGeneratorProvider.getRandomGenerator(8796864982253772625L);
+
+		Set<MultiKey> expectedObservations = new LinkedHashSet<>();
+		Set<MultiKey> actualObservations = new LinkedHashSet<>();
+		TestPluginData.Builder pluginBuilder = TestPluginData.builder();
+
+		// get testPropertyIds to use
+		ArrayList<TestPersonPropertyId> testPersonPropertyIds = new ArrayList<>();
+
+		for (TestPersonPropertyId testPersonPropertyId : TestPersonPropertyId.values()) {
+			if (testPersonPropertyId.getPropertyDefinition().propertyValuesAreMutable()) {
+				testPersonPropertyIds.add(testPersonPropertyId);
+			}
+		}
+
+		// set and subscribe to test actor plans for each testPropertyId
+		for (TestPersonPropertyId testPersonPropertyId : testPersonPropertyIds) {
+
+			// generate 50 random values
+			ArrayList<Object> allValues = new ArrayList<>();
+			for (int i = 0; i < 50; i++) {
+				Object value = testPersonPropertyId.getRandomPropertyValue(randomGenerator);
+				allValues.add(value);
+			}
+
+			// pick out unique values to subscribe to
+			ArrayList<Object> chosenValues = new ArrayList<>();
+			for (int i = 0; i < 50; i += 2) {
+				Object value = allValues.get(i);
+				if (chosenValues.contains(value)) {
+					chosenValues.add(allValues.get(i));
+				}
+			}
```

Comment:
What is this supposed to do?

You create a new list on line 1367 with no values and then here you are checking if that list contains the value. It won't, so chosenValues remains empty. If you ran the test and it passed, this is the reason it passed. If you add even one value to this, the test will fail.